core/src/layers/throttle.rs

```diff
@@ -0,0 +1,332 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use std::io::SeekFrom;
+use std::num::NonZeroU32;
+use std::sync::Arc;
+use std::task::{Context, Poll};
+
+use async_trait::async_trait;
+use bytes::Bytes;
+use chrono::Duration;
+use governor::clock::{Clock, DefaultClock, QuantaInstant};
+use governor::middleware::{RateLimitingMiddleware, StateInformationMiddleware, StateSnapshot};
+use governor::state::{InMemoryState, NotKeyed};
+use governor::{NegativeMultiDecision, NotUntil, Quota, RateLimiter, RatelimitedStream};
+
+use crate::raw::*;
+use crate::*;
+
+/// ThrottleLayer can help users to control the max bandwidth that used by OpenDAL.
+#[derive(Clone)]
+pub struct ThrottleLayer {
+    // TODO: NonZeroU32 or u32, since user enter 0 will cause panic
+    max_burst: u32,
+}
+
+impl ThrottleLayer {
+    /// Create a new ThrottleLayer will specify quota
+    pub fn new(max_burst: u32) -> Self {
+        Self { max_burst }
```

Comment:
If `max_burst` is set to zero, we should trigger a panic. Let's update the `Panics` section in the comments of the `new` function to reflect this behavior.

Reply:
Done